rpcs3/Crypto/ec.cpp

```diff
@@ -30,55 +30,49 @@ static int bn_compare(u8* a, u8* b, u32 n)
 	return 0;
 }
 
-static u8 bn_add_1(u8* d, u8* a, u8* b, u32 n)
+static inline u8 bn_add_1(u8* d, const u8* a, const u8* b, u32 n)
 {
-	u32 i;
-	u32 dig;
-	u8 c;
+	u8 c = 0;
 
-	c = 0;
-	for (i = n - 1; i < n; i--)
+	for (u32 i = n; i > 0; i--)
```

Comment:
I think either i != umax or check i == 0 after tge operation (before i--)

Reply:
i == 0 after operation would allow to not rely on overflow.

Reply:
I could cast to s64 which would probably do the same.
But what you suggest is also ill-defined.

Reply:
It's not, for (u32 i;; i--) { if (i == 0) break; }

Reply:
Turns out the whole ecdsa internally only uses n = 20 and n = 21.
So I made them template functions and simply use s32.